### PR TITLE
fix: Marvel example doesn't handle DioError in the UI

### PR DESCRIPTION
**Pull request by msarkrish**
I added a new variable to store the Dio exception message. If it is null, an error object will be displayed. This PR will close this [Issue](https://github.com/rrousselGit/riverpod/issues/167)

### CODE REVIEW

File: examples/marvel/lib/src/screens/home.dart [hunk 1]
```diff
@@ -89,10 +89,14 @@ class Home extends HookConsumerWidget {
...
           error: (err, stack) {
+            String? errorMessage;
+            if (err is DioException) {
+              errorMessage = err.message;
+            }
             return Scaffold(
               appBar: AppBar(title: const Text('Error')),
               body: Center(
-                child: Text('$err'),
+                child: Text(errorMessage ?? '$err'),
```
Comment (rrousselGit) [inline examples/marvel/lib/src/screens/home.dart:99]: Use Dart 3's switch instead. This would be more readable

Comment (msarkrish) [reply examples/marvel/lib/src/screens/home.dart]: Thank you for your review. Dart 3's switch is used now, Remi.

File: examples/marvel/lib/src/screens/home.dart [hunk 2]
```diff
@@ -96,7 +96,12 @@ class Home extends HookConsumerWidget {
             return Scaffold(
               appBar: AppBar(title: const Text('Error')),
               body: Center(
-                child: Text(errorMessage ?? '$err'),
+                child: Text(
+                  switch (errorMessage) {
+                    String() => errorMessage,
```
Comment (rrousselGit) [inline examples/marvel/lib/src/screens/home.dart:101]: I meant removing the `err is DioException` and that local `errorMessage` variable

Comment (msarkrish) [reply examples/marvel/lib/src/screens/home.dart]: Oh, I wrongly understood. Now I have updated the code, Remi.

File: examples/marvel/lib/src/screens/home.dart [hunk 3]
```diff
@@ -89,17 +89,13 @@ class Home extends HookConsumerWidget {
...
             return Scaffold(
               appBar: AppBar(title: const Text('Error')),
               body: Center(
                 child: Text(
-                  switch (errorMessage) {
-                    String() => errorMessage,
-                    null => '$err'
+                  switch (err) {
+                    DioException() => err.message ?? '$err',
+                    Object() => '$err',
```
Comment (rrousselGit) [inline examples/marvel/lib/src/screens/home.dart:98]:
```suggestion
                   _ => '$err',
```

**Comment by rrousselGit** [comment]
Thanks!

**Comment by msarkrish** [comment]
It's my pleasure to contribute to your package.

**Pull request merged**